test/embedded/deinit-release.swift

```diff
@@ -0,0 +1,26 @@
+// RUN: %empty-directory(%t)
+// RUN: %target-swift-frontend %s -enable-experimental-feature Embedded -O -c -o %t/main.o
+// RUN: %target-clang %t/main.o -o %t/a.out -dead_strip
+// RUN: %target-run %t/a.out | %FileCheck %s
+
+// REQUIRES: swift_in_compiler
+// REQUIRES: executable_test
+// REQUIRES: OS=macosx || OS=linux-gnu
+
+class C {}
+
+struct Foo {
+    let foo: C = C()
+    let bar: C = C()
+}
+
+class Bar {}
+class SubBar: Bar {
+    var qwe = Foo()
+}
+
+var bar: SubBar? = SubBar()
+bar = nil
+print("OK!")
+
+// CHECK: OK!
```

Comment:
I'm a bit lost as to how this test actually tests this scenario in question, and I worry it might silently stop testing this scenario with some compiler change. It might be better to use `Unmanaged` to do an explicit retain/release in a `deinit {}` block, or assign `self` to a global, call out to something opaque, then remove it.

Reply:
Added a more targeted test.

Reply:
Looks great, thanks!